types.go

```diff
@@ -46,7 +50,8 @@ type Config struct { // nolint:maligned
 	TLSEnable bool
 	CertKey   string
 	CertFile  string
-
+	// JSONMarshaller is used to marshal the response body to JSON.
+	JSONMarshaller JSONMarshaller
```

Comment:
Maybe, this field not need be an interface, could be more simple with something like this:

```go
JSONMarshal     func(v interface{}) ([]byte, error)
```